fix(amf): Malformed session accept message issue

Sathyaj27 opened the pull request:
Signed-off-by: Sathyaj27 <sathya.jayadev@wavelabs.ai>

<!--
    Tag your PR title with the components that it touches along with
    the type of change
    E.g. "fix(orc8r): Fix reindexer race condition" or "feat(agw) ..."
-->

## Summary
Fixing #12509 
Issue :

- Buffer allocation was wrong as the value was over written causing a minimum length failure in dnn even though DNN length was more than required value.

<!-- Enumerate changes you made and why you made them -->

## Test Plan
- Tested on abott.
- Testd Via UT.( TestPDUv6SessionSetup is the corresponding UT)
<img width="924" alt="neelima" src="https://user-images.githubusercontent.com/94469973/166926880-c6d11407-de57-44b8-b146-109e8eec806c.PNG">


## Additional Information

- [ ] This change is not backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

Sathyaj27 committed:
fix(amf): Malformed session accept message issue

Signed-off-by: Sathyaj27 <sathya.jayadev@wavelabs.ai>

panyogesh commented:
Looks Good.  Although not directly related to changes. But can you take care of lints.

Either 
  - guti_m5_t guti = {};  <<< No variable
Or
   guti_m5_t guti;  memset(&guti, 0, sizeof(guti_m5_t));

Sathyaj27 commented:
> Looks Good. Although not directly related to changes. But can you take care of lints.
> 
> Either
> 
> * guti_m5_t guti = {};  <<< No variable
>   Or
>   guti_m5_t guti;  memset(&guti, 0, sizeof(guti_m5_t));

Addressed the comment

ulaskozat reviewed (approved):
LGTM
